### DISCUSSION TITLE
a question about train_step

### DISCUSSION THREAD

**Discussion by minhoooo1** [Lightning-AI/pytorch-lightning#11989] [code help: CV]
I was reading other people's code and found PL, an excellent framework. But I have one doubt about the training step .Let me show you the code related to dataloader
![A](https://user-images.githubusercontent.com/61350752/154697931-c55c76cd-5ea4-46c4-b433-ad86b334e274.jpg)
![B](https://user-images.githubusercontent.com/61350752/154697973-79114510-bc86-45b7-aa52-2d232eba2364.jpg)

When I use 1000 training pictures, In the first figure len(self.files) = 1000 ,len(AEI_Dataset) = 5000 and batch_size = 20 ，train_ step should be 250, but it is 125000250 .You can see the figure below 。
![C](https://user-images.githubusercontent.com/61350752/154700265-808f09a2-dc3b-462c-9e0e-0273a693169e.jpg)

I hope a kind person can help me answer it. Thank you very much

**Comment by rohitgr7** [comment]
hey @minhoooo1 !

you can multiplied the total len by 5 inside your dataset's `__len__` method.
Also, the main progress bar shows the total steps (including both train & val) in the main progress bar. You can verify the total count by printing `len(dataloader)` to check the length.

**Comment by minhoooo1** [comment]
I tried, but the number of steps on the main process is different from&nbsp;len(dataloader)。I still don't know total steps&nbsp;calculation method at present. Do you have any documents? If one sentence is clear, it's best。Thank you！




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "PyTorchLightning/pytorch-lightning"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年2月21日(星期一) 下午4:20
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [PyTorchLightning/pytorch-lightning] a question about train_step (Discussion #11989)





 
hey @minhoooo1 !
 
you can multiplied the total len by 5 inside your dataset's __len__ method.
 Also, the main progress bar shows the total steps (including both train &amp; val) in the main progress bar. You can verify the total count by printing len(dataloader) to check the length.
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
